### PR TITLE
refactor(ui): project creation modal

### LINKED ISSUE #826: Keyboard shortcuts don't work as expected after navigating with subrouter
To reproduce:
1) Go to a `Project`
2) Click to Issues or Revisions tab
3) Try any keyboard shortcut (`modifierKey` + `,`; `modifierKey` + `p`, etc)

Modal will not appear

4) Now try clicking anywhere in the window and a keyboard shortcut - modal will appear

### LINKED ISSUE #835: Overhaul new project modal
- [x] use new floating style modal
- [x] change wording
- [x] add emoji
- [x] move repository selectors to the top
- [ ] Update the disabled field globally to include new background color
- [x] add tooltip to project name field when starting with an existing repo

### PR DESCRIPTION
**Pull request by MeBrei**
Closes #826 
Closes #835

### CODE REVIEW

**Review by rudolfs — COMMENTED** [review]
Looks good!

Some things I noticed:

- [ ] circular deps
```
(!) Circular dependencies
ui/src/modal.ts -> ui/Screen/ProjectCreation.svelte -> ui/DesignSystem/Component/index.js -> ui/DesignSystem/Component/ModalOverlay.svelte -> ui/src/modal.ts
ui/src/modal.ts -> ui/Screen/ProjectCreation.svelte -> ui/DesignSystem/Component/index.js -> ui/DesignSystem/Component/Sidebar.svelte -> ui/src/modal.ts
```
- [ ] lock screen during create request broken
- [ ] keyboard shortcuts don't work when opening the modal via the "new project" button (might be related to https://github.com/radicle-dev/radicle-upstream/issues/826)

**Comment by xla** [comment]
> ```
> (!) Circular dependencies
> ui/src/modal.ts -> ui/Screen/ProjectCreation.svelte -> ui/DesignSystem/Component/index.js -> ui/DesignSystem/Component/ModalOverlay.svelte -> ui/src/modal.ts
> ui/src/modal.ts -> ui/Screen/ProjectCreation.svelte -> ui/DesignSystem/Component/index.js -> ui/DesignSystem/Component/Sidebar.svelte -> ui/src/modal.ts
```

I ran into that as well when working on the navigation refactor. It might be good practice to separate the library code providing the machinery and the functionality mapping to actual components.

File: ui/Screen/Profile/ProjectsMenu.svelte [hunk 1]
```diff
@@ -1,15 +1,16 @@
 <script>
-  import { push } from "svelte-spa-router";
-
   import * as path from "../../src/path.ts";
+  import * as modal from "../../src/modal.ts";
```
Comment (xla) [inline ui/Screen/Profile/ProjectsMenu.svelte:3]:
```suggestion
  import * as modal from "../../src/modal.ts";
  import * as path from "../../src/path.ts";
```

**Review by xla — CHANGES_REQUESTED** [review]
I failed to bring this up during the original introduction of the new modal, but I don't see a reason why we have to hold onto the `path` concept even so we don't use a router anymore. An enum for our differrent modals could look like this:

``` ts
// app.ts

import { SvelteComponent } from "svelte";

import Help from "../Modal/Help.svelte";
import ProjectCreation from "../Modal/ProjectCreation.svelte";
import Shortcuts from "../Modal/Shortcuts.svelte";

export interface View<K extends string> {
  readonly component: unknown;
  readonly key: K,
  readonly props?: Props;
}

export type Props = Record<string, 0 | string>;
export type Map<Key extends string, C extends typeof SvelteComponent> = Required<
  Record<Key, C>
>;

export enum Modal {
  Help = "Help",
  ProjectCreation = "ProjectCreation",
  Shortcuts = "Shortcuts",
}

export const modalMap: Map<Modal, typeof SvelteComponent> = {
  [Modal.Help]: Help,
  [Modal.ProjectCreation]: ProjectCreation,
  [Modal.Shortcuts]: Shortcuts,
};
```

**Comment by MeBrei** [comment]
Closes #826 

3ec3d9a should also fix the issue above

**Comment by MeBrei** [comment]
> I failed to bring this up during the original introduction of the new modal, but I don't see a reason why we have to hold onto the `path` concept even so we don't use a router anymore. An enum for our differrent modals could look like this:
> ...

I refactored the setup of the modals in 511155f36c5cbfd56b57632611fa8b9e11b1bd3a
I kept the routes for now, this way, I like that everything is set up the same (ie routes map in `App.svelte` and the routes being accessible via `path.ts`). Having the mapping of the routes in `App.svelte` also avoids the circular dependencies. I also moved the modals into `Screen` as some of them have advanced functionality (eg API calls) that don't fit into `Component` and the modals basically form a screen in themselves.

**Comment by xla** [comment]
> I kept the routes for now, this way, I like that everything is set up the same (ie routes map in `App.svelte` and the routes being accessible via `path.ts`). Having the mapping of the routes in `App.svelte` also avoids the circular dependencies. I also moved the modals into `Screen` as some of them have advanced functionality (eg API calls) that don't fit into `Component` and the modals basically form a screen in themselves.

The consistency argument is a good one and I think we can drop the route concept once we migrate away from router based navigation altogether in #754.

I also believe that modals should be a first class concept and be treated like screens in the hierarchy and in their allowed capabilities/access. It feels strange to move them into the screen directory, I'd put them `ui/Modal`. So we can have:

```
./
../
ui/
ui/Modal/MyModal.svelte
ui/Modal/MyModal/Fragment1.svelte
ui/Modal/MyModal/Fragment2.svelte
ui/Screen/MyScreen.svelte
ui/Screen/MyScreen/Fragment1.svelte
ui/Screen/MyScreen/Fragment2.svelte
```

**Review by xla — APPROVED** [review]
🌙 🎁 🕧 🎸

**Comment by rudolfs** [comment]
Found some edge cases, that don't make sense after the reshuffling of the inputs in the form. Feel free to create a follow up issue if it's too much hassle to address in this PR.

- [ ] when choosing "create new repo", after selecting the directory path it should focus on the name field instead of immediately showing a validation error
Screenshot 2020-09-04 at 10 50 28

- [ ] when there's only one character in the name field, and the validations don't pass, the "create project" button is already enabled, it should stay disabled until all validations pass
Screenshot 2020-09-04 at 10 50 11

**Comment by MeBrei** [comment]
> Found some edge cases, that don't make sense after the reshuffling of the inputs in the form. Feel free to create a follow up issue if it's too much hassle to address in this PR.
> 

I think it would make sense to refactor the validation set up for this modal before addressing the edge cases, so I created a new issue #869

**Review by rudolfs — APPROVED** [review]

**Pull request merged**